source/TimePicker.js

```diff
@@ -628,35 +668,64 @@
 		*/
 		setChildPickers: function (inOld) {
 			if (this.value) {
-				var hour = this.value.getHours();
-				this.$.hour.setValue(hour);
-				this.$.minute.setValue(this.value.getMinutes());
-				if (this.meridiemEnable === true) {
-					this.$.meridiem.setValue(hour > 11 ? 1 : 0);
+				var values = this.calcPickerValues();
+				this.$.hour.set('value', values.hour);
+				this.$.minute.set('value', values.minute);
+				if (this.meridiemEnable === true) {					
+					this.$.meridiem.set('value', this.$.meridiem.getMeridiemIndex(values.hour, values.minute));
 				}
 			}
-			this.$.currentValue.setContent(this.formatValue());
+			this.$.currentValue.set('content', this.formatValue());
 		},
 
 		/**
 		* @private
 		*/
+		calcPickerValues: function () {
+			var values = {},
+				value = this.localeValue || this.value;
+
+			if (value) {
+				values.hour = value.getHours();
+				values.minute = value.getMinutes();
+			} else {
+				values.hour = values.minute = 0;
+			}
+
+			return values;
+		},
+
+		/**
+		* @private
+		*/
+		valueChanged: function (old) {
+			if (typeof ilib !== 'undefined') {
+				this.localeValue = ilib.Date.newInstance({unixtime: this.value.getTime(), timezone: 'local'});
```

Comment:
I think there is a case where we would change the value to `null` (as in to reset the `TimePicker`). Currently this would throw an error.